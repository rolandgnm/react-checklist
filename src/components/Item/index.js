import React from 'react'
import PropTypes from 'prop-types';

const Item = ({description, className, onClick}) => {
	let attributes = {
		onClick
	};

	if (className) {
		attributes.className = className;
	}
	return <li {...attributes}>
		{description}
	</li>;
};

Item.propTypes = {
	description: PropTypes.string.isRequired,
	onClick: PropTypes.func,
	className: PropTypes.string,
};

Item.defaultProps = {
	className: null,
	onClick: function () {}
};

export default Item