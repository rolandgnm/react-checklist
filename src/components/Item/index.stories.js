import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Item from '../Item/index';

import { withKnobs, text, boolean, number } from '@storybook/addon-knobs';


storiesOf('Item', module)
	.add('Basic List item', () => (
        <Item description="Description" />
	));