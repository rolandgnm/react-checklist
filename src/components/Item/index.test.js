import React from 'react';
import Item from './index';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('<Item />', () => {

	it('should render a single item', () => {
		const tree = renderer.create(
			<Item description="my first item" />
		).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render with a class name attribute', () => {
		const tree = renderer.create(
			<Item description="my first item" className="my-css-class" />
		).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should call a callback when the list item is click', () => {
		const myFunction = jest.fn();
		const wrapper = shallow(<Item description="My Item" onClick={myFunction} />);
		wrapper.simulate('click');
		expect(myFunction.mock.calls.length).toBe(1);
	});
});