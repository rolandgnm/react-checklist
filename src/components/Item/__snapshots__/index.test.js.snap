// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render a single item 1`] = `
<li
  onClick={[Function]}
>
  my first item
</li>
`;

exports[`<Item /> should render with a class name attribute 1`] = `
<li
  className="my-css-class"
  onClick={[Function]}
>
  my first item
</li>
`;
